relay/nakama/persona/verifier.go

```diff
@@ -88,56 +87,72 @@ func (p *Verifier) cleanupStaleEntries(now time.Time) {
+func (p *Verifier) handleReceipt(receipts []*receipt.Receipt) []string {
+	//nolint:prealloc // we cannot know how many receipts we're going to get from the dispatcher
+	var hashes []string
+	for _, rec := range receipts {
+		// Note: receiptConstant is the key returned in the result
+		// of the CreatePersonaResponse from Cardinal
+		result, ok := rec.Result[receiptConstant]
+		if !ok {
+			// Receipts that do not have the "success" key will be discarded here
+			continue
+		}
+		success, ok := result.(bool)
+		if !ok {
+			continue
+		}
+		pending := p.txHashToPending[rec.TxHash]
+		pending.lastUpdate = time.Now()
+		if success {
+			pending.status = StatusAccepted
+		} else {
+			pending.status = StatusRejected
+		}
+		p.txHashToPending[rec.TxHash] = pending
+		hashes = append(hashes, rec.TxHash)
 	}
-	p.txHashToPending[receipt.TxHash] = pending
-	return receipt.TxHash
+	return hashes
 }
 
-func (p *Verifier) handlePending(tuple txHashAndUserID) string {
+func (p *Verifier) handlePending(tuple txHashAndUserID) []string {
 	pending := p.txHashToPending[tuple.txHash]
 	pending.lastUpdate = time.Now()
 	pending.userID = tuple.userID
 	p.txHashToPending[tuple.txHash] = pending
-	return tuple.txHash
+	return []string{tuple.txHash}
 }
 
-func (p *Verifier) attemptVerification(txHash string) error {
-	pending, ok := p.txHashToPending[txHash]
-	if !ok || pending.userID == "" || pending.status == "" {
-		// We're missing a success/failure message from cardinal or the initial request from the
-		// user to claim a persona tag.
-		return nil
-	}
-	// We have both a user ID and a success message. Save this success/failure to nakama's storage system
-	ctx := context.Background()
-	ctx = context.WithValue(ctx, runtime.RUNTIME_CTX_USER_ID, pending.userID) //nolint:staticcheck // its fine.
-	ptr, err := LoadPersonaTagStorageObj(ctx, p.nk)
-	if err != nil {
-		return eris.Wrap(err, "unable to get persona tag storage obj")
-	}
-	if ptr.Status != StatusPending {
-		return eris.Errorf("expected a pending persona tag status but got %q", ptr.Status)
-	}
-	ptr.Status = pending.status
-	if err = ptr.SavePersonaTagStorageObj(ctx, p.nk); err != nil {
-		return eris.Wrap(err, "unable to set persona tag storage object")
+func (p *Verifier) attemptVerification(txHashes []string) error {
+	for _, txHash := range txHashes {
+		pending, ok := p.txHashToPending[txHash]
+		if !ok || pending.userID == "" || pending.status == "" {
+			// We're missing a success/failure message from cardinal or the initial request from the
+			// user to claim a persona tag.
+			return nil
+		}
+		// We have both a user ID and a success message. Save this success/failure to nakama's storage system
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, runtime.RUNTIME_CTX_USER_ID, pending.userID) //nolint:staticcheck // its fine.
+		ptr, err := LoadPersonaTagStorageObj(ctx, p.nk)
+		if err != nil {
+			return eris.Wrap(err, "unable to get persona tag storage obj")
```

Comment:
Now that this loops over a number of hashes, this should accumulate the errors in a slice and return them with a errors.Join. With this current state of the code, a problematic early item in txHashes will cause the rest of the verifications to fail.

e.g. If txHashes has 10 items to verify, and there's a problem with the 5th, items 6, 7, 8, and 9 won't be verified.